[bug] Panic when author is null

This appears to occur when the original author account no longer exists, and the API returns null.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x84fad66]

goroutine 12 [running]:
github.com/jimschubert/changelog/service.githubService.convertToChangeItem(0x980fd10, 0x98188a0, 0x986bfb0, 0x981e580, 0x9822cd0, 0x986bf88)
/go/src/app/service/github_service.go:108 +0x176
github.com/jimschubert/changelog/service.githubService.Process.func1(0x980fd10, 0x980fcf8, 0x980fd10, 0x98188a0, 0x981e580, 0x9822cd0, 0x998ea80, 0x998ea78, 0x9850300, 0x0, ...)
/go/src/app/service/github_service.go:80 +0x8c
created by github.com/jimschubert/changelog/service.githubService.Process
/go/src/app/service/github_service.go:77 +0x23a
```

--- a/service/github_service.go
+++ b/service/github_service.go
@@ -95,8 +95,16 @@ func (s githubService) convertToChangeItem(commit *github.RepositoryCommit, ch c
 		if !s.shouldExcludeViaRepositoryCommit(commit) {
 			excludeByGroup := false
 			var t *time.Time
-			if commit.GetCommit() != nil && (*commit.GetCommit()).GetAuthor() != nil && (*(*commit.GetCommit()).GetAuthor()).Date != nil {
-				t = (*(*commit.GetCommit()).GetAuthor()).Date
+			var authorRaw *string
+			var authorUrlRaw *string
+			if commit.GetCommit() != nil {
+				commitAuthor := *(*commit.GetCommit()).Author
+				d := commitAuthor.GetDate()
+				t = &d
+			}
+			if commit.Author != nil {
+				authorRaw = commit.Author.Login
+				authorUrlRaw = commit.Author.HTMLURL
 			}
 
 			grouping := (*s.config).FindGroup(commit.GetCommit().GetMessage())
@@ -105,8 +113,8 @@ func (s githubService) convertToChangeItem(commit *github.RepositoryCommit, ch c
 			if !excludeByGroup {
 				// TODO: Max count?
 				ci := &model.ChangeItem{
-					AuthorRaw:        commit.Author.Login,
-					AuthorURLRaw:     commit.Author.HTMLURL,
+					AuthorRaw:        authorRaw,
+					AuthorURLRaw:     authorUrlRaw,
 					CommitMessageRaw: commit.Commit.Message,
 					DateRaw:          t,
 					CommitHashRaw:    commit.SHA,

--- a/service/github_service_test.go
+++ b/service/github_service_test.go
@@ -15,9 +15,15 @@
 package service
 
 import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"sync"
 	"testing"
 
 	"github.com/google/go-github/v29/github"
+	"github.com/stretchr/testify/assert"
 
 	"github.com/jimschubert/changelog/model"
 )
@@ -87,6 +93,74 @@ func Test_githubService_shouldExcludeViaRepositoryCommit(t *testing.T) {
 			if got := s.shouldExcludeViaRepositoryCommit(tt.args.commit); got != tt.want {
 				t.Errorf("shouldExcludeViaRepositoryCommit() = %v, want %v", got, tt.want)
 			}
+		})
+	}
+}
+
+func commitFromJson(t *testing.T, file string) *github.RepositoryCommit {
+	t.Helper()
+	path := filepath.Join("testdata", file) // relative path
+	rawBytes, err := ioutil.ReadFile(path)
+	assert.NoError(t, err, "could not find %s", path)
+	var commit github.RepositoryCommit
+	err = json.Unmarshal(rawBytes, &commit)
+	assert.NoError(t, err, "could not unmarshal %s as github.RepositoryCommit", path)
+
+	return &commit
+}
+
+func Test_githubService_convertToChangeItem_authorInfo(t *testing.T) {
+	background := context.Background()
+	type fields struct {
+		contextual *Contextual
+		config     *model.Config
+	}
+	type args struct {
+		commit *github.RepositoryCommit
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		compare model.ChangeItem
+	}{
+		{
+			name:   "github api doc example commit",
+			fields: fields{newContextual(nil), &model.Config{}},
+			args:   args{commitFromJson(t,"commit.json") },
+			compare: model.ChangeItem{
+				AuthorRaw: github.String("octocat"),
+				AuthorURLRaw: github.String("https://github.com/octocat"),
+				CommitMessageRaw: github.String("Fix all the bugs"),
+			},
+		},
+		{
+			name:   "issue 1",
+			fields: fields{newContextual(nil), &model.Config{}},
+			args:   args{commitFromJson(t,"issue_1.json") },
+			compare: model.ChangeItem{
+				CommitMessageRaw: github.String("Fix all the bugs"),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := githubService{
+				contextual: tt.fields.contextual,
+				config:     tt.fields.config,
+			}
+
+			ciChan := make(chan *model.ChangeItem)
+			wg := sync.WaitGroup{}
+			go func() {
+				wg.Add(1)
+				s.convertToChangeItem(tt.args.commit, ciChan, &wg, &background)
+			}()
+			ci := <-ciChan
+			assert.NotNil(t, ci)
+			assert.Equal(t, tt.compare.Author(), ci.Author())
+			assert.Equal(t, tt.compare.AuthorURL(), ci.AuthorURL())
+			assert.Equal(t, github.Stringify(tt.compare.CommitMessageRaw), github.Stringify(ci.CommitMessageRaw))
 		})
 	}
 }

--- a/service/testdata/commit.json
+++ b/service/testdata/commit.json
@@ -1,0 +1,95 @@
+{
+  "url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+  "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
+  "node_id": "MDY6Q29tbWl0NmRjYjA5YjViNTc4NzVmMzM0ZjYxYWViZWQ2OTVlMmU0MTkzZGI1ZQ==",
+  "html_url": "https://github.com/octocat/Hello-World/commit/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+  "comments_url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e/comments",
+  "commit": {
+    "url": "https://api.github.com/repos/octocat/Hello-World/git/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+    "author": {
+      "name": "Monalisa Octocat",
+      "email": "mona@github.com",
+      "date": "2011-04-14T16:00:49Z"
+    },
+    "committer": {
+      "name": "Monalisa Octocat",
+      "email": "mona@github.com",
+      "date": "2011-04-14T16:00:49Z"
+    },
+    "message": "Fix all the bugs",
+    "tree": {
+      "url": "https://api.github.com/repos/octocat/Hello-World/tree/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+      "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e"
+    },
+    "comment_count": 0,
+    "verification": {
+      "verified": false,
+      "reason": "unsigned",
+      "signature": null,
+      "payload": null
+    }
+  },
+  "author": {
+    "login": "octocat",
+    "id": 1,
+    "node_id": "MDQ6VXNlcjE=",
+    "avatar_url": "https://github.com/images/error/octocat_happy.gif",
+    "gravatar_id": "",
+    "url": "https://api.github.com/users/octocat",
+    "html_url": "https://github.com/octocat",
+    "followers_url": "https://api.github.com/users/octocat/followers",
+    "following_url": "https://api.github.com/users/octocat/following{/other_user}",
+    "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
+    "starred_url": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
+    "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
+    "organizations_url": "https://api.github.com/users/octocat/orgs",
+    "repos_url": "https://api.github.com/users/octocat/repos",
+    "events_url": "https://api.github.com/users/octocat/events{/privacy}",
+    "received_events_url": "https://api.github.com/users/octocat/received_events",
+    "type": "User",
+    "site_admin": false
+  },
+  "committer": {
+    "login": "octocat",
+    "id": 1,
+    "node_id": "MDQ6VXNlcjE=",
+    "avatar_url": "https://github.com/images/error/octocat_happy.gif",
+    "gravatar_id": "",
+    "url": "https://api.github.com/users/octocat",
+    "html_url": "https://github.com/octocat",
+    "followers_url": "https://api.github.com/users/octocat/followers",
+    "following_url": "https://api.github.com/users/octocat/following{/other_user}",
+    "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
+    "starred_url": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
+    "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
+    "organizations_url": "https://api.github.com/users/octocat/orgs",
+    "repos_url": "https://api.github.com/users/octocat/repos",
+    "events_url": "https://api.github.com/users/octocat/events{/privacy}",
+    "received_events_url": "https://api.github.com/users/octocat/received_events",
+    "type": "User",
+    "site_admin": false
+  },
+  "parents": [
+    {
+      "url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+      "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e"
+    }
+  ],
+  "stats": {
+    "additions": 104,
+    "deletions": 4,
+    "total": 108
+  },
+  "files": [
+    {
+      "filename": "file1.txt",
+      "additions": 10,
+      "deletions": 2,
+      "changes": 12,
+      "status": "modified",
+      "raw_url": "https://github.com/octocat/Hello-World/raw/7ca483543807a51b6079e54ac4cc392bc29ae284/file1.txt",
+      "blob_url": "https://github.com/octocat/Hello-World/blob/7ca483543807a51b6079e54ac4cc392bc29ae284/file1.txt",
+      "patch": "@@ -29,7 +29,7 @@\n....."
+    }
+  ]
+}

--- a/service/testdata/issue_1.json
+++ b/service/testdata/issue_1.json
@@ -1,0 +1,76 @@
+{
+  "url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+  "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
+  "node_id": "MDY6Q29tbWl0NmRjYjA5YjViNTc4NzVmMzM0ZjYxYWViZWQ2OTVlMmU0MTkzZGI1ZQ==",
+  "html_url": "https://github.com/octocat/Hello-World/commit/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+  "comments_url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e/comments",
+  "commit": {
+    "url": "https://api.github.com/repos/octocat/Hello-World/git/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+    "author": {
+      "name": "Monalisa Octocat",
+      "email": "mona@github.com",
+      "date": "2011-04-14T16:00:49Z"
+    },
+    "committer": {
+      "name": "Monalisa Octocat",
+      "email": "mona@github.com",
+      "date": "2011-04-14T16:00:49Z"
+    },
+    "message": "Fix all the bugs",
+    "tree": {
+      "url": "https://api.github.com/repos/octocat/Hello-World/tree/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+      "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e"
+    },
+    "comment_count": 0,
+    "verification": {
+      "verified": false,
+      "reason": "unsigned",
+      "signature": null,
+      "payload": null
+    }
+  },
+  "author": null,
+  "committer": {
+    "login": "octocat",
+    "id": 1,
+    "node_id": "MDQ6VXNlcjE=",
+    "avatar_url": "https://github.com/images/error/octocat_happy.gif",
+    "gravatar_id": "",
+    "url": "https://api.github.com/users/octocat",
+    "html_url": "https://github.com/octocat",
+    "followers_url": "https://api.github.com/users/octocat/followers",
+    "following_url": "https://api.github.com/users/octocat/following{/other_user}",
+    "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
+    "starred_url": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
+    "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
+    "organizations_url": "https://api.github.com/users/octocat/orgs",
+    "repos_url": "https://api.github.com/users/octocat/repos",
+    "events_url": "https://api.github.com/users/octocat/events{/privacy}",
+    "received_events_url": "https://api.github.com/users/octocat/received_events",
+    "type": "User",
+    "site_admin": false
+  },
+  "parents": [
+    {
+      "url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+      "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e"
+    }
+  ],
+  "stats": {
+    "additions": 104,
+    "deletions": 4,
+    "total": 108
+  },
+  "files": [
+    {
+      "filename": "file1.txt",
+      "additions": 10,
+      "deletions": 2,
+      "changes": 12,
+      "status": "modified",
+      "raw_url": "https://github.com/octocat/Hello-World/raw/7ca483543807a51b6079e54ac4cc392bc29ae284/file1.txt",
+      "blob_url": "https://github.com/octocat/Hello-World/blob/7ca483543807a51b6079e54ac4cc392bc29ae284/file1.txt",
+      "patch": "@@ -29,7 +29,7 @@\n....."
+    }
+  ]
+}

